polyfill/lib/duration.mjs

```diff
@@ -565,6 +565,39 @@ export class Duration {
     }
     return ES.ToTemporalDuration(item, this);
   }
+  static compare(one, two, options = undefined) {
+    one = ES.ToTemporalDuration(one, Duration);
+    two = ES.ToTemporalDuration(two, Duration);
+    options = ES.NormalizeOptionsObject(options);
+    const relativeTo = ES.ToRelativeTemporalObject(options);
+    const y1 = GetSlot(one, YEARS);
+    const mon1 = GetSlot(one, MONTHS);
+    const w1 = GetSlot(one, WEEKS);
+    let d1 = GetSlot(one, DAYS);
+    const h1 = GetSlot(one, HOURS);
+    const min1 = GetSlot(one, MINUTES);
+    const s1 = GetSlot(one, SECONDS);
+    const ms1 = GetSlot(one, MILLISECONDS);
+    const µs1 = GetSlot(one, MICROSECONDS);
+    let ns1 = GetSlot(one, NANOSECONDS);
+    const y2 = GetSlot(two, YEARS);
+    const mon2 = GetSlot(two, MONTHS);
+    const w2 = GetSlot(two, WEEKS);
+    let d2 = GetSlot(two, DAYS);
+    const h2 = GetSlot(two, HOURS);
+    const min2 = GetSlot(two, MINUTES);
+    const s2 = GetSlot(two, SECONDS);
+    const ms2 = GetSlot(two, MILLISECONDS);
+    const µs2 = GetSlot(two, MICROSECONDS);
+    let ns2 = GetSlot(two, NANOSECONDS);
+    if (y1 !== 0 || y2 !== 0 || mon1 !== 0 || mon2 !== 0 || w1 !== 0 || w2 !== 0 || d1 !== 0 || d2 !== 0) {
+      ({ days: d1 } = ES.UnbalanceDurationRelative(y1, mon1, w1, d1, 'days', relativeTo));
```

Comment:
Is there a different PR that's going to make this code work with ZDT in the case where there's a DST transition during the days portion?  I assume so because the ZDT-specific tests are skipped and this code won't adjust for DST, but wanted to make sure I wasn't missing something obvious.

Reply:
Sorry, I should have copied this from the commit message into the PR description: "Allowing ZonedDateTime as relativeTo does not work yet." I intend to change all the call sites that accept relativeTo in a follow-up, as the last change that will tie both #569 and #856 together.

Reply:
I've managed to get this to work with ZonedDateTime today, and updated the pull request.